### PR TITLE
ISSUE-674 JMAP API proposal for deleted message vault

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: docs/modules/ROOT/pages/tmail-backend/jmap-extensions/deletedMessagesVault.adoc [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+ - `waiting`: the task is planned and waits to be executed
+ - `inProgress`: the task is currently being executed
+ - `done`: the task is done. It's execution did succeed.
+ - `failed`: the task execution failed.
+ - `canceled`: the task was canceled.
+
+=== EmailRecoveryAction/get
+
+Standard `/get` methods. the `ids` property cannot be null.
```
Comment (quantranhong1999) [inline docs/modules/ROOT/pages/tmail-backend/jmap-extensions/deletedMessagesVault.adoc:49]: `the `ids` property cannot be null.` 
-> Issue: how can the client track the submitted task between diff devices/ or even diff browsers?

Comment (chibenwa) [reply docs/modules/ROOT/pages/tmail-backend/jmap-extensions/deletedMessagesVault.adoc:49]: Answer: Not supported.

Comment (chibenwa) [inline docs/modules/ROOT/pages/tmail-backend/jmap-extensions/deletedMessagesVault.adoc:49]:
```suggestion
Standard `/get` methods. The `ids` property cannot be null.
```

Comment (chibenwa) [inline docs/modules/ROOT/pages/tmail-backend/jmap-extensions/deletedMessagesVault.adoc:49]:
```suggestion
Standard `/get` methods. The `ids` property cannot be null.
```

**Review by quantranhong1999 — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Pull request merged**